nixos/tests/nvmetcfg.nix

```diff
@@ -0,0 +1,43 @@
+import ./make-test-python.nix ({ lib, ... }: {
+  name = "nvmetcfg";
+
+  meta = {
+    maintainers = with lib.maintainers;[ nickcao ];
+  };
+
+  nodes = {
+    server = { pkgs, ... }: {
+      boot.kernelModules = [ "nvmet" ];
+      environment.systemPackages = [ pkgs.nvmetcfg ];
+      networking.firewall.allowedTCPPorts = [ 8009 ];
```

Comment:
I see you've chosen a different port, the one for the discovery.

After reading the specs, I think this is slightly less "correct".
If you look at the [NVM Express TCP Transport Specification](https://nvmexpress.org/wp-content/uploads/NVM-Express-TCP-Transport-Specification-1.0d-2023.12.27-Ratified.pdf), there is one important line in block 3.1.2:
> NVMe/TCP I/O controllers should not use TCP port 8009

The other port standardized is 4420 (which I use in my examples) - it is just declared as "NVM Express over Fabrics storage access" on the [IANA assignment site](https://www.iana.org/assignments/service-names-port-numbers/service-names-port-numbers.xhtml?search=nvme).

Since - at least to my knowledge - it is impossible to create a discovery-only controller in the linux nvmet subsystem and they must share a port, 4420 seems the most appropriate.